deploy/helm/elastic-agent/templates/agent/_helpers.tpl

```diff
@@ -0,0 +1,412 @@
+{{- end -}}
+{{- end -}}
+{{- if empty $.Values.agent.fleet.preset -}}
+{{- fail "preset must be defined when fleet mode is enabled" -}}
+{{- end -}}
+{{- if not (hasKey $.Values.agent.presets $.Values.agent.fleet.preset)}}
+{{- fail (printf "specified preset \"%s\" under fleet is not found" $.Values.agent.fleet.preset) -}}
+{{- end -}}
+{{- range $presetName, $presetVal := $.Values.agent.presets}}
+{{- if ne $presetName $.Values.agent.fleet.preset -}}
+{{- $_ := unset $.Values.agent.presets $presetName}}
+{{- end -}}
+{{- end -}}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Validate and initialise outputs
+*/}}
+{{- define "elasticagent.init.outputs" -}}
+{{- $ := . -}}
+{{- $supportOutputTypes := dict "ESPlainAuth" "" "ESSecretAuthBasic" "" "ESSecretAuthAPI" "" "ESECKRef" ""}}
+{{- range $outputName, $outputVal := $.Values.outputs -}}
+{{- if empty ($outputVal).type -}}
+{{- $_ := set $outputVal "type" "ESPlainAuth" -}}
+{{- end -}}
+{{- include (printf "elasticagent.output.%s.validate" ($outputVal).type) (list $ $outputName $outputVal)}}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Initialise input templates if we are not deploying as managed
+*/}}
+{{- define "elasticagent.init.inputs" -}}
+{{- $ := . -}}
+{{- if eq $.Values.agent.fleet.enabled false -}}
+{{/* standalone agent so initialise inputs */}}
+{{- include "elasticagent.kubernetes.init" $ -}}
+{{- range $customInputName, $customInputVal := $.Values.extraIntegrations -}}
+{{- $customInputPresetName := ($customInputVal).preset -}}
+{{- $_ := required (printf "customInput \"%s\" is missing required preset field" $customInputName) $customInputPresetName -}}
+{{- $presetVal := get $.Values.agent.presets $customInputPresetName -}}
+{{- $_ := required (printf "preset with name \"%s\" of customInput \"%s\" not defined" $customInputPresetName $customInputName) $customInputVal -}}
+{{- $customInputOuput := dig "use_output" (list) $customInputVal -}}
+{{- if empty $customInputOuput -}}
+{{- fail (printf "output not defined in custom integration \"%s\"" $customInputName) -}}
+{{- end -}}
+{{- include "elasticagent.preset.mutate.outputs.byname" (list $ $presetVal $customInputOuput) -}}
+{{- include "elasticagent.preset.mutate.inputs" (list $ $presetVal (list $customInputVal)) -}}
+{{- end -}}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Validate and initialise the defined agent presets
+*/}}
+{{- define "elasticagent.init.presets" -}}
+{{- $ := . -}}
+{{- range $presetName, $presetVal := $.Values.agent.presets -}}
+{{- include "elasticagent.preset.mutate.unprivileged" (list $ $presetVal) -}}
+{{- include "elasticagent.preset.mutate.fleet" (list $ $presetVal) -}}
+{{- $presetMode := dig "mode" ("") $presetVal -}}
+{{- if not $presetMode -}}
+{{- fail (printf "mode is missing from preset \"%s\"" $presetName) -}}
+{{- else if eq $presetMode "deployment" -}}
+{{- else if eq $presetMode "statefulset" -}}
+{{- else if eq $presetMode "daemonset" -}}
```

Comment:
sure thing, if Helm provided a way to automate this it would be way easier but as of now this is too much extra work and I would propose to be part of a follow up PR